### PR TITLE
context FEATURE allow custom content-id in yang-library data

### PR DESCRIPTION
**Pull request by michalvasko**

### CODE REVIEW

File: src/context.c [hunk 1]
```diff
@@ -322,10 +322,11 @@ ly_ctx_unset_options(struct ly_ctx *ctx, uint16_t option)
 }
 
 API uint16_t
-ly_ctx_get_module_set_id(const struct ly_ctx *ctx)
+ly_ctx_get_change_count(const struct ly_ctx *ctx)
```
Comment (rkrejci) [inline src/context.c:325]: please add the information about the change into the transition manual

File: src/context.h [hunk 2]
```diff
@@ -510,24 +508,27 @@ uint32_t ly_ctx_internal_modules_count(const struct ly_ctx *ctx);
...
+ * with the key value "complete".
+ *
+ * If the data identifier can be limited to the existence and changes of this context, the following
+ * last 2 parameters can be used:
+ *
+ * "%u" as @p content_id_format and ::ly_ctx_get_change_count() as its parameter.
+ *
  * @param[in] ctx Context with the modules.
  * @param[out] root Generated yang-library data.
+ * @param[in] content_id_format Format string for the yang-library data identifier, which is the "content_id" node
```
Comment (rkrejci) [inline src/context.h:526]: please mention explicitely that it is printf's format string

File: tests/utests/basic/test_yanglib.c [hunk 3]
```diff
@@ -114,7 +114,7 @@ test_yanglib(void **state)
     UTEST_ADD_MODULE(schema_a, LYS_IN_YANG, feats, NULL);
     UTEST_ADD_MODULE(schema_b, LYS_IN_YANG, NULL, NULL);
 
-    assert_int_equal(LY_SUCCESS, ly_ctx_get_yanglib_data(UTEST_LYCTX, &tree));
+    assert_int_equal(LY_SUCCESS, ly_ctx_get_yanglib_data(UTEST_LYCTX, &tree, "%u", ly_ctx_get_change_count(UTEST_LYCTX)));
```
Comment (rkrejci) [inline tests/utests/basic/test_yanglib.c:117]: please add also some weird format strings, like `"<<%u>>"`, `"%u" , -10`, to make sure that these situations are handled correctly (from the code I believe they are).

Comment (rkrejci) [reply tests/utests/basic/test_yanglib.c:117]: or just `""`.

**Review by rkrejci — CHANGES_REQUESTED** [review]

File: src/context.h (hunk 2 above)
Comment (michalvasko) [reply src/context.h]: Fixed.

File: tests/utests/basic/test_yanglib.c (hunk 3 above)
Comment (michalvasko) [reply tests/utests/basic/test_yanglib.c:123]: The implementation should be completely dependent on `vasprintf` functionality so I considered more tests redundant. In any case, the ones you mentioned were added.

**Pull request merged**